generators/entity-server/templates/dotnetcore/src/Project/Service/Dto.cs.ejs

```diff
@@ -0,0 +1,135 @@
+            <%_  } else if (relationshipType === 'many-to-many') {
+                const joinEntityNamePascalized = relationships[idx].joinEntityNamePascalized ;
+                const joinEntityFieldNamePascalizedPlural = relationships[idx].joinEntityFieldNamePascalizedPlural ;
+                const ownerSide = relationships[idx].ownerSide; _%>
+        [JsonIgnore]
+        public IList<<%= joinEntityNamePascalized %>> <%= joinEntityFieldNamePascalizedPlural %> { get; } = new List<<%= joinEntityNamePascalized %>>();
+
+        [NotMapped]
+                <%_ if (!ownerSide) { _%>
+        [JsonIgnore]
+                <%_ } _%>
+        public IList<<%= asDto(otherEntityNamePascalized) %>> <%= relationshipFieldNamePascalizedPlural %> { get; }
+            <%_ } _%>
+
+        <%_ } _%>
+        // jhipster-needle-entity-add-field - JHipster will add fields here, do not remove
+
+        <%_ if (hasManyToMany) { _%>
+        public <%= entityClassName %>()
+        {
+            <%_ for (idx in relationships) {
+                const relationshipType = relationships[idx].relationshipType;
+                if(relationshipType === 'many-to-many'){
+                    const relationshipFieldNamePascalizedPlural = relationships[idx].relationshipFieldNamePascalizedPlural;
+                    const otherEntityNamePascalized = relationships[idx].otherEntityNamePascalized;
+                    const joinEntityNamePascalized = relationships[idx].joinEntityNamePascalized ;
+                    const joinEntityFieldNamePascalizedPlural = relationships[idx].joinEntityFieldNamePascalizedPlural ; _%>
+            <%= relationshipFieldNamePascalizedPlural %> = new JoinListFacade<<%= asDto(otherEntityNamePascalized) %>, <%= asDto(entityClassName) %>, <%= joinEntityNamePascalized %>>(this, <%= joinEntityFieldNamePascalizedPlural %>);
+                <%_ }
+            } _%>
+        }
+
+        <%_ } _%>
+        public override bool Equals(object obj)
+        {
+            if (this == obj) return true;
+            if (obj == null || GetType() != obj.GetType()) return false;
+            var <%= entityVariableName %> = obj as <%= asDto(entityClassName) %>;
+            if (<%= entityVariableName %>?.Id == null || <%= entityVariableName %>?.Id == 0 || Id == 0) return false;
+            return EqualityComparer<long>.Default.Equals(Id, <%= entityVariableName %>.Id);
+        }
+
+        public override int GetHashCode()
+        {
+            return HashCode.Combine(Id);
+        }
+
+        public override string ToString()
+        {
+            return "<%= entityClassName %>{" +
+                    $"ID='{Id}'" +
+            <%_ for (idx in fields) {
+                const fieldNamePascalized = fields[idx].fieldNamePascalized; _%>
+                    $", <%= fieldNamePascalized %>='{<%= fieldNamePascalized %>}'" +
+            <%_ } _%>
+                    "}";
+        }
```

Comment:
The DTO need this part ?

Reply:
I think it can be removed since this code was from the entity class